oeps/processes/oep-0055-proc-project-maintainers.rst

```diff
@@ -0,0 +1,428 @@
+meeting the requirements of that program.
+
+The term **maintainer scrum-of-scrums** will mean the group of all
+maintainers who are responsible for flagging issues to project
+leadership and proposing process improvements for component
+maintenance.
+
+The term **Technical Oversight Committee** (TOC) will mean the group
+responsible for the overall oversight and stewardship of the Open edX
+project. The TOC's function with respect to the project is to be the
+arbiter of last resort for governance matters. The TOC itself was established
+and is governed by a formal
+charter.  You can read the full document `here
+<https://openedx.org/wp-content/uploads/2021/12/TCRIL-Technical-Oversight-Committee-Charter-rev.11-16.21.pdf>`_.
+
+Historical Context
+------------------
+Historically, the Open edX project has not had a well-defined role
+of maintainer.  At times, community members stepped into this role
+informally.  The role was also part of a broader "owner" role at edX.
+Owners were also responsible for the health of the component they
+owned both in the repository and in production; owners answered the
+pager when something went wrong.
+
+What Maintainers Do
+-------------------
+The role of the repository or project maintainer is both a community
+focused and a technical role [#role]_.  Maintainers will be the public face
+of each component and should be welcoming and available.  While
+maintainers should strive to always be friendly and kind in their
+communication, firmness and directness are equally important.
+Maintainers must enforce the project's standards.  In cases where a
+proposed change isn't aligned with the our standards or direction, it
+is best to make that clear as early as possible.
+
+Non-technical responsibilities of maintainers include:
+
+* Effective communication with community members, other maintainers,
+  and project leadership
+* Selection and approval of Core Contributors with commit rights to their repositories
+* Disciplined management to ensure that maintenance work is done
+* Enforcement of the project's standards, both technical and social: maintainers
+  must remain friendly, kind, and open, but that should not supercede
+  enforcement of technical standards
+* The ability to solve or escalate issues facing users of the
+  component
+
+Technical responsibilities of maintainers include:
+
+* Understanding the current component architecture and implementation
+* Ensuring that basic documenation for the component exists and is
+  accurate
+* Having a documented technical vision for the evolution of the
+  component
+* The ability to assess the appropriateness of proposed changes to or
+  extensions of the component
+* The role of the component in the overall platform and how it
+  interacts with other components
+* Must be available to participate in the comment period for Core Contributor
+  nominees nominated to commit code to the maintainer's repo(s), as part of the
+  `Core Contributor Program`_
+  
```

Comment:
The section on technical responsibilities doesn't explicitly mention a few things that come to mind when I think of "maintainer" responsibilities:
* Something like "ensure ongoing compatibility with other components of the Open edX ecosystem as they are developed" (e.g. changes to edx-platform, or Paragon, or even new Ubuntu LTS versions, etc.). This could be part of "Having a documented technical vision for the evolution of the component"?
* Something like "ensure dependencies are updated as needed to avoid security risks or use of components that unsupported/unmaintained/end-of-life."

I suppose those could fall under "Disciplined management to ensure that maintenance work is done" and "enforcement of technical standards" from the non-technical section? Either way if they're expected I think calling them out explicitly is helpful.

Reply:
@bradenmacdonald this has now been added:

```
* Ensure that dependencies are updated as needed to avoid security risks or use
  of components that or not fit for use, e.g., unsupported, not maintained, or
  at their end-of-life.
```

To this point:

> Something like "ensure ongoing compatibility with other components of the Open edX ecosystem as they are developed" (e.g. changes to edx-platform, or Paragon, or even new Ubuntu LTS versions, etc.). This could be part of "Having a documented technical vision for the evolution of the component"?

I'm also not sure where to put this or how to phrase this. I could see simply putting your quote under Technical Responsibilities. Do you have a suggestion/opinion?

Reply:
Putting it under technical responsibilities seems fine to me!